package-testing/sdk-test-runner/src/app.ts

```diff
@@ -0,0 +1,149 @@
+import axios from 'axios';
+import config from './config';
+import * as fs from 'fs';
+import path from 'path';
+import { green, log, logIndent, red, yellow } from './logging';
+import { AssignmentRequest } from './dto/assignmentRequest';
+import { first, isEqual } from 'lodash';
+import { BanditActionRequest } from './dto/banditActionRequest';
+import { sleep } from './util';
+import { TestResponse } from './dto/testResponse';
+
+log(`Firing up test runner for ${config.sdkName}`);
+log(`Posting test cases to SDK server at ${config.sdkServer}`);
+log(`Controlling configuration server at ${config.apiServer}`);
+
+const testConfig = JSON.parse(fs.readFileSync(config.scenarioFile, 'utf-8'));
+
+const scenarios = testConfig['scenarios'];
+
+const failures: TestResponse[] = [];
+let numTestCases = 0;
+// failures.push({error: "I'm a fake failure"});
+
+let firstTestCase = true;
+const testScenario = async (key: string) => {
+
+    const safeSdkName = encodeURIComponent(config.sdkName);
+
+    await axios
+        .post(`${config.apiServer}/sdk/${safeSdkName}/scenario`, { label: key })
+        .then((result) => {
+            if (result.status != 200) {
+                log(yellow(`API server returned an error when setting test scenario`));
+                console.log(result);
+                // STOP
+            }
+            log(`Set Testing Scenario to ${key}`);
+        })
+        .catch((error) => {
+            console.error('Error:', error);
+            // STOP
+        });
+
+    if (!firstTestCase) {
+        log('Sleeping between scenarios so SDK can reload cache');
+        await sleep(31000);
+    } else {
+        firstTestCase = false;
+    }
+
+    log(`Loading test files for scenario, ${key}`);
+
+    const testCaseDir = path.join(config.testDataPath, scenarios[key]['testCases']);
+    const testCases = fs.readdirSync(testCaseDir);
```

Comment:
we should throw an error if testCases is empty, as that would indicate something is wrong (yet the test would "pass")